src/other/tables.jl

```diff
@@ -32,7 +32,11 @@ function DataFrame(x::T; copycols::Bool=true) where {T}
     return fromcolumns(Tables.columns(x), copycols=copycols)
 end
 
-Base.append!(df::DataFrame, x) = append!(df, DataFrame(x, copycols=false))
+Base.append!(df::DataFrame, table; cols::Symbol=:setequal) =
+    append!(df, DataFrame(x, copycols=false), cols=cols)
+Base.append!(df::DataFrame, table::Dict; cols::Symbol=:setequal) =
+    throw(ArgumentError("passing `Dict` as `table` when `cols` is equal to " *
```

Comment:
Wait, the error must be thrown only when `cols == :orderequal`, right?

Reply:
Sure! Thank you for catching this. Fixed and added to tests